src/components/AddNewComment.jsx

```diff
@@ -0,0 +1,61 @@
+import { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { addCommentToArticleByID } from "../utils/api";
+import { UserContext } from "../contexts/UserContext";
+import { Link } from "react-router-dom";
+
+function AddNewComment({ comments, setComments }) {
+  const { user } = useContext(UserContext);
+  const [newComment, setNewComment] = useState({ username: user, body: "" });
+  const [error, setError] = useState(false);
+
+  const { article_id } = useParams();
+
+  const handleChange = (event) => {
+    setNewComment((newComment) => {
+      const edit = { ...newComment };
+      edit.body = event.target.value;
+
+      return edit;
+    });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    addCommentToArticleByID(article_id, newComment)
+      .then((response) => {
+        setComments([response, ...comments]);
+      })
+
+      .catch(() => {
+        setError(true);
+      });
+  };
+
```

Comment:
You should reset the content of your comment box to empty after a comment is posted, for ease of use, and it would mean the user isn't able to mash the post button and send loads of comments (something that might be quite likely user behaviour if the user has slow internet)